analysis/src/topObjectSelection.cc

```diff
@@ -115,13 +118,16 @@ vector<TParticle> topObjectSelection::jetSelection() {
     jet.SetMomentum(mom);
     jet.SetFirstMother(i);
     jets.push_back(jet);
-    for (UInt_t iu = 0; iu < 19; iu++) {
-     // Jet_SF_CSV[iu] *= m_btagSF.getSF(jet, Jet_btagCSVV2[i], Jet_hadronFlavour[i], iu);
-    }
+    b_btagCSVV2 = Jet_btagCSVV2[i];
+    //BTagEntry::JetFlavor JF = BTagEntry::FLAV_UDSG;
+    //BTagEntry::JetFlavor JF;
+    if (abs(Jet_hadronFlavour[i]) == 5) JF = BTagEntry::FLAV_B;
+    //else if (abs(Jet_hadronFlavour[i]) == 4) JF = BTagEntry::FLAV_C;
+    auto bjetSF = m_btagSF.eval_auto_bounds("central", JF , Jet_eta[i], Jet_pt[i], Jet_btagCSVV2[i]);
```

Comment:
what happens when its not a b jet? it looks like the scale factor code will choose randomly before the first b jet, and always give bjet afterward